src/cargo/core/compiler/custom_build.rs

```diff
@@ -208,7 +208,7 @@ fn build_work<'a, 'cfg>(cx: &mut Context<'a, 'cfg>, unit: &Unit<'a>) -> CargoRes
 
     // Be sure to pass along all enabled features for this package, this is the
     // last piece of statically known information that we have.
-    for feat in &unit.features {
+    for &feat in &unit.features {
```

Comment:
Can you explain these changes to add `&`?  It looks more confusing to me.

Reply:
The IDE suggested that there's double reference in one of those variables (like &&str), so adding & helps avoiding it. Then I looked for similar patterns and changed them as well.

On the second thought it's not very important, so I have removed that commit.